config/config.go

```diff
@@ -351,29 +354,60 @@ func GetDefaultConfigPath(workingDir string) string {
 func FindConfigFilesInPath(rootPath string, terragruntOptions *options.TerragruntOptions) ([]string, error) {
 	configFiles := []string{}
 
-	err := filepath.Walk(rootPath, func(path string, info os.FileInfo, err error) error {
+	// Start from the given root path, unless we're given specific directories to include or strict include is on.
+	directories := []string{rootPath}
+	if terragruntOptions.StrictInclude || len(terragruntOptions.IncludeDirs) > 0 {
+		directories = []string{}
+	}
+
+	// Build a collection of directories based off of any included directories from the given terragrunt options.
+	for _, directory := range terragruntOptions.IncludeDirs {
+		matches, err := zglob.Glob(filepath.Join(rootPath, directory))
 		if err != nil {
-			return err
+			return nil, err
 		}
 
-		// Skip the Terragrunt cache dir entirely
-		if info.IsDir() && info.Name() == options.TerragruntCacheDir {
-			return filepath.SkipDir
-		}
+		directories = append(directories, matches...)
+	}
 
-		isTerragruntModule, err := containsTerragruntModule(path, info, terragruntOptions)
-		if err != nil {
-			return err
-		}
+	// TODO: since the terragrunt include directories can be specified as globs, we need to expand
+	// those globs before walking over them.
```

Comment:
This TODO is now addressed in the for loop above right? If so, we should remove this.

Reply:
Ah yes, forgot to take that out. Good catch.